yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -208,6 +219,24 @@ export class AztecNodeService implements AztecNode {
     return this.merkleTreeDB.getSiblingPath(MerkleTreeId.PRIVATE_DATA_TREE, leafIndex, false);
   }
 
+  /**
+   * Find the index of the relevant l1 to l2 message.
+   * @param leafValue - The value to search for.
+   * @returns The index of the given leaf in the l1 to l2 message tree or undefined if not found.
+   */
+  public findL1ToL2MessageIndex(leafValue: Buffer): Promise<bigint | undefined> {
```

Comment:
Right now can this be merged with getL1ToLeMessage below so that getL1ToL2Message returns the message and an index.